src/blister.h

```diff
@@ -209,29 +209,70 @@ static inline UInt * BLOCKS_BLIST(Obj list)
 /****************************************************************************
 **
 *F  COUNT_TRUES_BLOCK( <block> )  . . . . . . . . . . . count number of trues
+**
+**  This function uses either the gcc builtin __builtin_popcount which
+**  generates the popcntl or popcntq instruction on sufficiently recent intel
+**  CPUs or some alternative code otherwise, or the algorithm below
+**
+**  The sequence to compute the  number of bits in  a block is quite  clever.
+**  The idea is that after the <i>-th instruction each subblock of $2^i$ bits
+**  holds the number of   bits of this  subblock  in the original block  <m>.
+**  This is illustrated in the example below for a block of with 8 bits:
+**
+**       // a b c d e f g h
+**      m = (m & 0x55)       +  ((m >> 1) & 0x55);
+**       // . b . d . f . h  +  . a . c . e . g   =  a+b c+d e+f g+h
+**      m = (m & 0x33)       +  ((m >> 2) & 0x33);
+**       // . . c+d . . g+h  +  . . a+b . . e+f   =  a+b+c+d e+f+g+h
+**      m = (m & 0x0f)       +  ((m >> 4) & 0x0f);
+**       // . . . . e+f+g+h  +  . . . . a+b+c+d   =  a+b+c+d+e+f+g+h
+**
+**  In the actual  code  some unnecessary mask  have  been removed, improving
+**  performance quite a bit,  because masks are 32  bit immediate values  for
+**  which most RISC  processors need two  instructions to load them.  Talking
+**  about performance.  The code is  close to optimal,  it should compile  to
+**  only about  22 MIPS  or SPARC instructions.   Dividing the  block into  4
+**  bytes and looking up the number of bits  of a byte in a  table may be 10%
+**  faster, but only if the table lives in the data cache.
+**
```

Comment:
Why is just this paragraph formatted in block text? (I personally prefer to just do left-adjusted.

Reply:
Goes back to Martin. I'm not sure why it shows up here as an addition.

Reply:
And why is this duplicated between source and header file? I know, I know, we do this in lots of places -- and I always thought it's a bad idea, because the two copies will either drift apart, or cause extra work when one needs to tweak them.

So I'd just keep one copy -- probably this one here, right next to the code.